### PR TITLE
set media entry to fix Undefined array key

### LINKED ISSUE #3958: "Undefined array key "media" in inc/ChangeLog/RevisionInfo.php on line 39" in 2023-04-04 "Jack Jackrum"
Hi,

Since I upgraded to 2023-04-04 "Jack Jackrum" and I know observe php warnings when I'm prompted to recover a draft

**Steps to reproduce**
- From an artbitrary page of your wiki: edit a page, make some changes, make sure the changes are autosaved, and then close your tab without saving your changes (ie: make sure there is an open draft for that page)
- Go back to that page and select `Edit draft`

**Expected behavior**
I expect to land on the page `Draft file found` and that it shows no error or warnings

**Actual behavior**
The page `Draft file found` has the expected content but also the warning `Warning: Undefined array key "media" in /xxx/inc/ChangeLog/RevisionInfo.php on line 39`

The stacktrace is:
```
{main}( )	.../doku.php:0
act_dispatch( )	.../doku.php:126
include( '/var/www/html/wikiDev/lib/tpl/dokuwiki/main.php )	.../actions.php:27
tpl_content( $prependTOC = ??? )	.../main.php:59
dokuwiki\Extension\Event::createAndTrigger( $name = 'TPL_ACT_RENDER', $data = 'draft', $action = 'tpl_content_core', $canPreventDefault = ??? )	.../template.php:85
dokuwiki\Extension\Event->trigger( $action = 'tpl_content_core', $enablePrevent = TRUE )	.../Event.php:199
[call_user_func_array:{/xxx/inc/Extension/Event.php:133}](http://www.php.net/function.call-user-func-array:%7B/var/www/html/wikiDev/inc/Extension/Event.php:133%7D)( $callback = 'tpl_content_core', $args = [0 => 'draft'] )	.../Event.php:133
tpl_content_core( 'draft' )	.../Event.php:133
dokuwiki\Action\Draft->tplContent( )	.../template.php:100
dokuwiki\Ui\PageDraft->show( )	.../Draft.php:40
dokuwiki\Ui\PageDiff->compareWith( $text = '<nspages >' )	.../PageDraft.php:35
dokuwiki\ChangeLog\RevisionInfo->__construct( $info = ['date' => 1675800528, 'ip' => '127.0.0.1', 'type' => 'E', 'id' => 'sandbox:00-index', 'user' => 'guillaume', 'sum' => '', 'extra' => '', 'sizechange' => -46] )
```

### PR DESCRIPTION
**Pull request by Klap-in**
Fix #3958

### CODE REVIEW

**Comment by dregad** [comment]
Thanks @Klap-in ! 👍

File: inc/Ui/PageDiff.php [hunk 1]
```diff
@@ -66,7 +66,9 @@ public function compareWith($text = null)
             $changelog =& $this->changelog;
 
             // revision info object of older file (left side)
-            $this->RevInfo1 = new RevisionInfo($changelog->getCurrentRevisionInfo());
+            $info = $changelog->getCurrentRevisionInfo();
+            $info['media'] = false;
```
Comment (splitbrain) [inline inc/Ui/PageDiff.php:70]: I don't like this. If having media set is important, why doesn't getCurrentRevisionInfo() return it? Looking into that, why doesn't getRevisionInfo() return it? Finally why do these functions return an array which is then used to initialize a RevisionInfo object? Why do these functions not return the initialized objects?

**Review by splitbrain — COMMENTED** [review]
Adding the media info should be moved at least higher up the chain as a quick fix. Ultimately this needs more refactoring.

**Comment by Klap-in** [comment]
RevisionInfo is mainly used for the Ui components.
getRevisionInfo() returns an array from the parsed log lines, used at more places, also in plugins.

So returning RevisionInfo objects for getRevisionInfo is not backward compatible.

**Comment by dregad** [comment]
Possibly fixes #3970 too (didn't check)

**Pull request merged**